apps/authentication-service/src/shared/users/usersService.ts

```diff
@@ -0,0 +1,97 @@
+import { JSONApiResponse, UserInfoClient, UserInfoResponse } from "auth0";
+
+import envs from "@/config";
+import { getAccessToken } from "@/shared/auth/authService";
+import { User } from "@/shared/users/usersModel";
+import { parseResponse } from "@/shared/utils";
+
+const authDomain = envs.AUTH0_DOMAIN;
+const userInfo: UserInfoClient = new UserInfoClient({
+  domain: envs.AUTH0_DOMAIN,
+  clientInfo: {
+    scopes: ["openid", "profile", "email", "offline_access"],
+    name: envs.AUTH0_CLIENT_ID,
+  },
+});
+
+const getUserInfoFromToken = async (
+  token: string,
+): Promise<JSONApiResponse<UserInfoResponse>> => {
+  return await userInfo.getUserInfo(token);
+};
+
+export const getUserIdFromToken = async (token: string): Promise<string> => {
+  const user: JSONApiResponse<UserInfoResponse> =
+    await getUserInfoFromToken(token);
+  return Promise.resolve(user.data.sub);
+};
+
+export const getUserClientId = async (userId: string): Promise<string> => {
+  const userInfo: User = await getUserById(userId);
+  return Promise.resolve(userInfo?.user_metadata?.client_id || "");
+};
+
+export const getUserById = async (userId: string): Promise<User> => {
+  const token = await getAccessToken();
+  const userIdEncoded = encodeURIComponent(userId);
+
+  return await parseResponse<User>(
+    await fetch(`https://${authDomain}/api/v2/users/${userIdEncoded}`, {
+      method: "GET",
+      headers: {
+        Accept: "application/json",
+        Authorization: `Bearer ${token}`,
+        "Content-Type": "application/json",
+      },
+    }),
+  );
+};
+
+// Keeping for future functionality
+export const getUsersByClientId = async (clientId: string): Promise<User[]> => {
```

Comment:
We should remove it if it's not used. This adds scope and increases the amount of things we have to test.

Reply:
I removed it.